plugins/ingest-geoip/src/main/java/org/elasticsearch/ingest/geoip/GeoIpProcessor.java

```diff
@@ -64,14 +65,30 @@
 
     private final String field;
     private final String targetField;
-    private final DatabaseReader dbReader;
+    private final DatabaseReaderLazyLoader dbReader;
```

Comment:
maybe rename `dbReader` to `dbLoader`?

Reply:
This does mean the lazy loader class will have to be opened up. The reason for this is because in some tests we don’t read from a dike on disk but from an embedded resource. That’s different than reading a file, so for that to work I have to add hooks to allow that test to override how to get the size of the stream and how to open a stream to that resource. That’s why the class is no longer. I went down this path and then reverted it all, and forgot to revert the non-final change.

Reply:
gotcha, thanks for explaining.

Reply:
This comment should have been a thread on [this review comment]. Continuing here, I pushed this change. Do you want to take a look?